src/zmqsubscriber/zmqsubscriber_test.go

```diff
@@ -1,32 +1,39 @@
 package zmqsubscriber
 
 import (
+	"log"
+	"testing"
+	"time"
+
 	"github.com/0xb10c/bademeister-go/src/test"
-	"github.com/0xb10c/bademeister-go/src/types"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"log"
-	"testing"
 )
 
-func captureTxs(z *ZMQSubscriber) *[]types.Transaction {
-	txs := []types.Transaction{}
-	go func() {
-		for t := range z.IncomingTx {
-			txs = append(txs, t)
-		}
-	}()
-	return &txs
+// waitForZMQTransaction waits for a new transaction to come in over ZMQ.
+// It returns with `didTimeout == false` as soon as a transactions comes in.
+// Otherwise it times out after the specified `timeout` and returns `didTimeout == true`.
+func waitForZMQTransaction(t *testing.T, z *ZMQSubscriber, timeout time.Duration) (didTimeout bool) {
+	select {
+	case <-z.IncomingTx:
+		return false
+	case <-time.After(timeout):
+		t.Logf("Timed out while waiting for a transaction (timeout %s)", timeout)
+		return false
```

Comment:
`return true` according to the docs

however I think returning `true` (and `assert.IsTrue`) if something was received is more natural

Reply:
actually in future tests we probably want to look at the deserialized value so maybe this should be `&tx`/`nil` instead (with `assert.NotNil()`)

Reply:
Good point! Addressed.